src/NBitcoin.Tests/TestUtils.cs

```diff
@@ -32,27 +20,28 @@ internal static byte[] ParseHex(string data)
             return Encoders.Hex.DecodeData(data);
         }
 
-        public static Block CreateFakeBlock(Transaction tx)
+        public static Block CreateFakeBlock(Network network, Transaction tx)
         {
-            var block = new Block();
+            Block block = network.CreateBlock();
             block.AddTransaction(tx);
             block.UpdateMerkleRoot();
             return block;
         }
 
-        public static Block CreateFakeBlock()
+        public static Block CreateFakeBlock(Network network)
         {
-            Block block = CreateFakeBlock(new Transaction());
+            Block block = CreateFakeBlock(network, network.CreateTransaction());
```

Comment:
you can just call `CreateFakeBlock(this.network);` here.

Reply:
Not with this one, as you will end up recursively calling the same method? There is a `public static Block CreateFakeBlock(Network network, Transaction tx)` above.

Reply:
ah yes lol, what was I thinking.